lib/spack/spack/environment/environment.py

```diff
@@ -2634,6 +2634,36 @@ def _ensure_env_dir():
 
     shutil.copy(envfile, target_manifest)
 
+    # Copy relative path includes that live inside the environment dir
+    try:
+        manifest = EnvironmentManifestFile(environment_dir)
+    except Exception:
+        # error handling for bad manifests is handled on other code paths
+        return
+
+    includes = manifest[TOP_LEVEL_KEY].get("include", [])
+    for include in includes:
+        if os.path.isabs(include):
+            continue
+
+        abspath = pathlib.Path(os.path.normpath(environment_dir / include))
+        common_path = pathlib.Path(os.path.commonpath([environment_dir, abspath]))
+        if common_path != environment_dir:
+            # Warn that we are not copying relative path
+            msg = "Spack will not copy relative include path from outside environment"
+            msg += f" directory: {include}"
+            tty.warn(msg)
+            continue
+
+        orig_abspath = os.path.normpath(envfile.parent / include)
+        if not os.path.exists(orig_abspath):
+            msg = f"Spack will not copy relative include path for nonexistent file: {include}"
+            tty.warn(msg)
```

Comment:
Likewise here -- you might have an env where some of the included files are generated or who knows what.  I would not warn by default.

I think more useful might be an option to fail, e.g., `--fail-missing-includes`, so that anyone who cares could check it.